### ISSUE TITLE
n_jobs opens a limited number of processes

### ISSUE THREAD

**Issue by PartiallyTyped** [SimonBlanke/Hyperactive#18]
**Describe the bug**
n_jobs does not start the provided number of processes, instead it starts `min(n_jobs, 12)`, and with `n_jobs=-1` it runs 12 jobs.

**Code to reproduce the behavior**

```python
points = np.random.rand(100)

def model(opt):
    return np.sum(points)

hyper = Hyperactive(distribution="joblib")
space = dict(a=np.arange(10))
hyper.add_search(model, space, n_iter=50, n_jobs=20)
hyper.run()
# len(hyper.results_list) shows 12
```

**Error message from command line**
NA

**System information:**
 - OS Platform and Distribution: MacOS 10.16.4
 - Python version: 3.8.5
 - Hyperactive version: 3.0.5

**Additional context**

Basically, I want to run the optimization algorithm e.g. 20  times and as far as I understood, I could achieve that with n_jobs, but n_jobs creates fewer runs.

**Comment by SimonBlanke** [comment]
Hello @PartiallyTyped,

I did some reading about multiprocessing with more processes than cores. I think you are right that this should be possible.

**Comment by SimonBlanke** [comment]
I fixed this bug in 390874f and tested it in e6fbd0c. The fix can be downloaded with version 3.0.5.1. Please let me know if this works for you.

**Comment by PartiallyTyped** [comment]
Hi, thanks Simon, it appears that it is working. Would it be possible to have `n_jobs` active while running `n_experiments`. 

Basically, I need to run my experiment, including the optimization process 500 times, but spawning 500 python processes is very ram intensive. So would it be possible to schedule N experiments to be run while having M active processes? 

I have experience with `multiprocessing` and `joblib` so I would be glad to submit a patch for this.

**Comment by SimonBlanke** [comment]
Hello @PartiallyTyped,

what would `n_experiments` be in this scenario? 
As far as i understood you want to run 100 processes in 5 experiments instead of 500 processes all at once, to avoid excessive ram usage. 

- What is stopping you running those 5 experiments one after another? You can collect all the results after an optimization run has finished and pass it to the next one.

 - There is also the possibility to do parallel computing within the objective function.

 - If you want to tinker with the parallel computing in Hyperactive you can do so via a wrapper function.

You can pass this wrapper function to "distribution". It will be used instead of the default. This is what this function would look like (from the API reference):
```python
def multiprocessing_wrapper(process_func, search_processes_paras, **kwargs):
  n_jobs = len(search_processes_paras)

  pool = Pool(n_jobs, **kwargs)
  results = pool.map(process_func, search_processes_paras)

  return results
```

If you have a patch for an option to make Hyperactive deal with hundreds of processes more efficiently, i would like to include it.

**Comment by SimonBlanke** [comment]
This issue was fixed by b89548894a86ac3232759dca9523c80181c77894 to work with the current API. Thanks again to @PartiallyTyped for suggesting this feature and contributing to it.
